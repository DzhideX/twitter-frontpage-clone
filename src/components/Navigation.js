import React from 'react';
import twitter from '../images/twitter.png';
import hashtag from '../images/navigation-icons/hashtag.png';
import birdhouse from '../images/navigation-icons/birdhouse.png';
import bell from '../images/navigation-icons/bell.png';
import bookmark from '../images/navigation-icons/bookmark.png';
import envelope from '../images/navigation-icons/envelope.png';
import list from '../images/navigation-icons/list.png';
import more from '../images/navigation-icons/more.png';
import headshot1 from '../images/headshot1.jpg';


const NavigationDiv = ({imgsrc, text}) => {
  return (
      <div className='container'>
        <img className={imgsrc===headshot1 ? 'headshot1' : null} src={imgsrc} height='27px' width='27px' alt=''/>
        <p> {text} </p>
      </div>
  );
};

const Navigation = () => {
    return (
        <div className='navigation'>
                <div className='buttons-container'>
                    <img className='tweet-logo' src={twitter} height='45px' width='45px' alt=''/>
                    <NavigationDiv imgsrc={birdhouse} text={'Home'}/>
                    <NavigationDiv imgsrc={hashtag} text={'Explore'}/>
                    <NavigationDiv imgsrc={bell} text={'Notifications'}/>
                    <NavigationDiv imgsrc={envelope} text={'Messages'}/>
                    <NavigationDiv imgsrc={bookmark} text={'Bookmarks'}/>
                    <NavigationDiv imgsrc={list} text={'Lists'}/>
                    <NavigationDiv imgsrc={headshot1}text={'Profile'}/>
                    <NavigationDiv imgsrc={more} text={'More'}/>
                    <div className='button-container'>
                        <button> Tweet </button>
                    </div>
            </div>
        </div>
    );
};

export { Navigation as default };