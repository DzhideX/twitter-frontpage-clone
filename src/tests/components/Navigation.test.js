import React from 'react';
import { shallow } from 'enzyme';
import Navigation from '../../components/Navigation';
import {createSerializer} from 'enzyme-to-json';

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));


it('renders without crashing', () => {
  const wrapper = shallow(<Navigation />);
  expect(wrapper).toMatchSnapshot();
});