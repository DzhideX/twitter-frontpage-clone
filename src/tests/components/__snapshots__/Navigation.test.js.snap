// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="navigation"
>
  <div
    className="buttons-container"
  >
    <img
      alt=""
      className="tweet-logo"
      height="45px"
      src="twitter.png"
      width="45px"
    />
    <NavigationDiv
      imgsrc="birdhouse.png"
      text="Home"
    />
    <NavigationDiv
      imgsrc="hashtag.png"
      text="Explore"
    />
    <NavigationDiv
      imgsrc="bell.png"
      text="Notifications"
    />
    <NavigationDiv
      imgsrc="envelope.png"
      text="Messages"
    />
    <NavigationDiv
      imgsrc="bookmark.png"
      text="Bookmarks"
    />
    <NavigationDiv
      imgsrc="list.png"
      text="Lists"
    />
    <NavigationDiv
      imgsrc="headshot1.jpg"
      text="Profile"
    />
    <NavigationDiv
      imgsrc="more.png"
      text="More"
    />
    <div
      className="button-container"
    >
      <button>
         Tweet 
      </button>
    </div>
  </div>
</div>
`;
